GreedyParameterPointSampler.C

```diff
@@ -0,0 +1,1420 @@
+        sprintf(tmp, "convergence_subset_size");
+        database.getInteger(tmp, d_convergence_subset_size);
+        sprintf(tmp, "next_point_to_sample");
+        database.getInteger(tmp, d_next_point_to_sample);
+        sprintf(tmp, "next_point_requiring_error_indicator");
+        database.getInteger(tmp, d_next_point_requiring_error_indicator);
+        sprintf(tmp, "check_local_rom");
+        database.getInteger(tmp, bool_int_temp);
+        d_check_local_rom = (bool) bool_int_temp;
+        sprintf(tmp, "use_centroid");
+        database.getInteger(tmp, bool_int_temp);
+        d_use_centroid = (bool) bool_int_temp;
+        sprintf(tmp, "iteration_started");
+        database.getInteger(tmp, bool_int_temp);
+        d_iteration_started = (bool) bool_int_temp;
+        sprintf(tmp, "convergence_started");
+        database.getInteger(tmp, bool_int_temp);
+        d_convergence_started = (bool) bool_int_temp;
+        sprintf(tmp, "next_parameter_point_computed");
+        database.getInteger(tmp, bool_int_temp);
+        d_next_parameter_point_computed = (bool) bool_int_temp;
+        sprintf(tmp, "point_requiring_error_indicator_computed");
+        database.getInteger(tmp, bool_int_temp);
+        d_point_requiring_error_indicator_computed = (bool) bool_int_temp;
+        sprintf(tmp, "subset_created");
+        database.getInteger(tmp, bool_int_temp);
+        d_subset_created = (bool) bool_int_temp;
+        sprintf(tmp, "debug_algorithm");
+        database.getInteger(tmp, bool_int_temp);
+        d_debug_algorithm = (bool) bool_int_temp;
+        sprintf(tmp, "counter");
+        database.getInteger(tmp, d_counter);
+        sprintf(tmp, "subset_counter");
+        database.getInteger(tmp, d_subset_counter);
+        sprintf(tmp, "random_seed");
+        database.getInteger(tmp, d_random_seed);
+
+        sprintf(tmp, "parameter_point_random_indices");
+        d_parameter_point_random_indices.resize(d_parameter_points.size());
+        database.getIntegerArray(tmp, &d_parameter_point_random_indices[0], d_parameter_points.size());
+
+        sprintf(tmp, "parameter_point_errors");
+        d_parameter_point_errors.resize(d_parameter_points.size());
+        database.getDoubleArray(tmp, &d_parameter_point_errors[0], d_parameter_points.size());
+
+        sprintf(tmp, "parameter_point_local_rom");
+        d_parameter_point_local_rom.resize(d_parameter_points.size());
+        database.getIntegerArray(tmp, &d_parameter_point_local_rom[0], d_parameter_points.size());
+
+        for (int i = 0; i < d_convergence_subset_size; i++)
+        {
+            std::string vec_path = base_file_name + "_conv_" + std::to_string(i);
+            Vector point;
+            point.read(vec_path);
+            d_convergence_points.push_back(point);
+        }
+    }
+    database.close();
+
+    rng.seed(d_random_seed + d_parameter_sampled_indices.size());
+}
+
+void
+GreedyParameterPointSampler::constructParameterPoints()
```

Comment:
This function just checks some conditions and prints out information, whereas `constructParameterPoints` in the derived classes actually constructs points. It seems better to rename this as something about checking input rather than constructing points. Also, that would avoid having the same name for different functions (not virtual).